### PR TITLE
Use ControllerBase for scaffolded API controllers

### PR DESCRIPTION
**Pull request by seancpeters**
Puts scaffolding in alignment with templating, per #636

### CODE REVIEW

**Comment by prafullbhosale** [comment]
Was this PR meant to be against the `dev` or `release/2.1` branch?

File: src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiControllerWithActions.cshtml [hunk 1]
```diff
@@ -11,7 +11,8 @@ namespace @Model.NamespaceName
 {
     [Produces("application/json")]
```
Comment (danroth27) [inline src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiControllerWithActions.cshtml]: We should remove the `ProducesAttribute`

File: src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiControllerWithContext.cshtml [hunk 2]
```diff
@@ -25,7 +25,8 @@ namespace @Model.ControllerNamespace
 }
     [Produces("application/json")]
```
Comment (danroth27) [inline src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiControllerWithContext.cshtml]: Remove `ProducesAttribute`

File: src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiEmptyController.cshtml [hunk 3]
```diff
@@ -10,7 +10,8 @@ namespace @Model.NamespaceName
 {
     [Produces("application/json")]
```
Comment (danroth27) [inline src/VS.Web.CG.Mvc/Templates/ControllerGenerator/ApiEmptyController.cshtml]: Remove `ProducesAttribute`

**Comment by seancpeters** [comment]
@prafullbhosale - It should be release/2.1 - thanks (fixed).

**Comment by seancpeters** [comment]
@danroth27 - the "extra" stuff was because I had the PR against the wrong branch. That's fixed now, and this now only has changes to the API controllers.

**Comment by seancpeters** [comment]
@danroth27 - can you take another look at this PR, now that I've cleaned up the extra stuff (It was there because I had the original PR pointed at the wrong branch).

**Pull request merged**